File: Java/1557_Minimum_Number_of_Vertices_to_Reach_All_Nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        List<Integer> result = new ArrayList<>();
        boolean[] t = new boolean[n];
        
        for(int i = 0; i < edges.size(); i++){
            t[edges.get(i).get(1)] = true;
        }
        
        for(int i = 0; i < n; i++){
            if(t[i] == false){
                result.add(i);
            }
        }
        return result;
    }
}

File: CS/1557_Minimum_Number_of_Vertices_to_Reach_All_Nodes.cs
public class Solution {
    public IList<int> FindSmallestSetOfVertices(int n, IList<IList<int>> edges) {
        IList<int> result = new List<int>();
        HashSet<int> s = new HashSet<int>();
        for(int i = 0; i < edges.Count; i++){
            s.Add(edges[i][1]);
        }
        for(int i = 0; i < edges.Count; i++){
            if(s.Add(edges[i][0])){
                result.Add(edges[i][0]);
            }
        }
        return result;
    }
}
